apps/heureka/src/components/filters/FilterSelect.jsx

```diff
@@ -6,51 +6,78 @@
 import React, { useState } from "react"
 import { Button, InputGroup, SelectOption, Select, Stack, SearchInput } from "@cloudoperators/juno-ui-components"
 import {
-  useFilterLabels,
-  useFilterLabelValues,
+  useIssueMatchesFilterLabels,
+  useServiceFilterLabels,
+  useComponentFilterLabels,
+  useIssueMatchesFilterLabelValues,
+  useServiceFilterLabelValues,
+  useComponentFilterLabelValues,
   useFilterActions,
-  useActiveFilters,
-  useSearchTerm,
+  useIssueMatchesActiveFilters,
+  useServiceActiveFilters,
+  useComponentActiveFilters,
+  useIssueMatchesSearchTerm,
+  useServiceSearchTerm,
+  useComponentSearchTerm,
 } from "../../hooks/useAppStore"
 import { humanizeString } from "../../lib/utils"
 
+const entityHooks = {
```

Comment:
Same here. Why not simply pass filterLabels, filterLabelValues, and activeFilters as props to the component? This would eliminate the extra logic inside the component, making it easier to test.